Item Alternate Image Features in e-commerce site.

Solufyin opened the pull request:
Hello @nabinhait
Added **Item Alternate Image Features** in e-commerce site.
We request to you closed previous pull request #15015.
As per your changed request #15015 We have developed Item Alternate Image Features.
We keep both features in single doctype. Slideshow and Item alternate image functionality shown in e-commerce site.

**The User can use two alternative choices:**

**1) If user is select slideshow option in slideshow drop down then user is able to see **slideshow functionality** in e-commerce site.**
![11](https://user-images.githubusercontent.com/34390782/43381406-328934f4-93f2-11e8-8d86-f353aa934d2f.png)
![12](https://user-images.githubusercontent.com/34390782/43381416-36678544-93f2-11e8-917b-502094d2a07e.png)

**2) If user is not select slideshow option in slideshow drop down then a user is able to see **Item Alternate Image** functionality in e-commerce site.**
![13](https://user-images.githubusercontent.com/34390782/43381437-44f6e24e-93f2-11e8-9618-9f719ebbaf1e.png)
![14](https://user-images.githubusercontent.com/34390782/43381441-48706dd2-93f2-11e8-8e80-a646e68323ed.png)
![15](https://user-images.githubusercontent.com/34390782/43381447-4b1a096c-93f2-11e8-8613-2560ae92dcfb.png)


**We configured images in slideshow item doctype in below snapshot:**

![con](https://user-images.githubusercontent.com/34390782/43381455-54a56724-93f2-11e8-9e58-7e870f77acb8.png)


**Please find attached video for more information:**

[item_alternate_image.mp4.zip](https://github.com/frappe/erpnext/files/2239954/item_alternate_image.mp4.zip)


Thanks.

ketanserpentcs committed:
[ADD] Added Item Alternate Image Features in e-commerce site.

netchampfaris commented on erpnext/public/css/website.css:
```diff
@@ -268,3 +268,21 @@
   margin-bottom: 20px;
 }
+/* For item alternate image */
+div.list_all_images img:hover {
```
Put these styles in website.less

netchampfaris commented on erpnext/templates/generators/item.html:
```diff
@@ -3,122 +3,152 @@
+                </div>
+            </div>
+         </div>
+        <div class="row item-website-description" style="margin-top:30px; margin-bottom:20px">
```
Put these inline styles in website.less ?

netchampfaris reviewed (changes requested):
I think we should replace the Image Slideshow altogether and set this mode as default.

ketanserpentcs committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into solufy_item_alternate_image

ketanserpentcs committed:
[IMP] Improved Code for Item Alternate Image Feature.

Solufyin commented:
Dear @netchampfaris ,

We Improved code as per your suggestion.

netchampfaris commented:
@Solufyin Hey, as I mentioned in my review, we should remove the old item slideshow and replace it with the new browsing mode you have added. It should become the default mode of browsing multiple images of the item.

gmplab commented:
i think Solufyin's solution is better. The user can make their own decisions, use the new way or the old way.

Solufyin commented:
@gmplab, "i think Solufyin's solution is better. The user can make their own decisions, use the new mode or the old mode. You can set the new mode as default."

Yes, The user can make their own choice that's why we keep both functionality slide show and item alternate image in eCommerce site.

ketanserpentcs committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into solufy_item_alternate_image

ketanserpentcs committed:
[IMP] Improved code for Item alternative image functionality.

Solufyin commented:
Dear @netchampfaris , 

"@Solufyin Hey, as I mentioned in my review, we should remove the old item slideshow and replace it with the new browsing mode you have added. It should become the default mode of browsing multiple images of the item."

We removed old slideshow and added new item alternate image functionality set as default mode as per your review.

netchampfaris committed:
Remove .css file and make a build for erpnext-web.css

netchampfaris committed:
Cleanup styling for item alternate image, also add delegated handler

netchampfaris committed:
Spaces -> Tabs

netchampfaris committed:
Spaces -> Tabs in item.html

netchampfaris commented:
![item-alternative-image](https://user-images.githubusercontent.com/9355208/43768422-f0b5e28e-9a54-11e8-843c-44d7c0cd1cec.gif)

generare commented:
This look awesome! All ok with it now?

generare commented:
Hello again! We were checking the Alternative Images functionality and it will be absolutely nice to have this functionality but it seems to have slightly many steps to input the alternate images if understood correctly. It would be nice if we would not need to jump to a Slideshow interface but rather upload the images on the same item interface page.

We borrowed the images from @Solufyin and edited those a bit to show what we mean.
![alternate-images-backend](https://user-images.githubusercontent.com/5984026/51556705-9a501780-1e83-11e9-826f-198e7ffdb2e0.jpg)
